app/src/backend/HiapService.ts

```diff
@@ -0,0 +1,466 @@
+  if (!contextData) throw new Error("No city context/emissions data found");
+
+  // Use HiapApiService
+  const { taskId } = await startPrioritization(contextData);
+  if (!taskId) throw new Error("No taskId returned from HIAP API");
+  const ranking = await db.models.HighImpactActionRanking.create({
+    locode,
+    inventoryId,
+    lang,
+    jobId: taskId,
+    status: HighImpactActionRankingStatus.PENDING,
+  });
+  // Do not await here, it will make the request time out. Poll job in the background.
+  checkActionRankingJob(ranking, lang, type);
+  return ranking;
+};
+
+async function fetchAndMergeRankedActions(
+  lang: LANGUAGES,
+  rankedActions: { actionId: string; rank: number; explanation: any; type: ACTION_TYPES }[],
+) {
+  const allActions = await GlobalAPIService.fetchAllClimateActions(lang);
+  
+  return rankedActions.map((rankedAction) => {
+    const details = allActions.find((a: any) => a.ActionID === rankedAction.actionId);
+    if (!details) {
+      logger.error(`No action details found for ActionID: ${rankedAction.actionId}`);
+      return null;
+    }
+    
+    return {
+      ...rankedAction,
+      explanation: rankedAction.explanation,
+      name: details.ActionName,
+      hazard: details.Hazard,
+      sector: details.Sector,
+      subsector: details.Subsector,
+      primaryPurpose: details.PrimaryPurpose,
+      description: details.Description,
+      cobenefits: details.CoBenefits,
+      equityAndInclusionConsiderations: details.EquityAndInclusionConsiderations,
+      GHGReductionPotential: details.GHGReductionPotential,
+      adaptationEffectiveness: details.AdaptationEffectiveness,
+      costInvestmentNeeded: details.CostInvestmentNeeded,
+      timelineForImplementation: details.TimelineForImplementation,
+      dependencies: details.Dependencies,
+      keyPerformanceIndicators: (details.KeyPerformanceIndicators),
+      powersAndMandates: (details.PowersAndMandates),
+      adaptationEffectivenessPerHazard: details.AdaptationEffectivenessPerHazard,
+      biome: details.biome,
+    };
+  }).filter((r) => r !== null);
+}
+
+// Helper: Save ranked actions for a language
+async function saveRankedActionsForLanguage(
+  ranking: HighImpactActionRanking,
+  rankedActions: any[],
+  lang: LANGUAGES,
+) {
+  const updatedRanking = await db.models.HighImpactActionRanking.findByPk(ranking.id);
+  if (updatedRanking?.status === HighImpactActionRankingStatus.PENDING) {
+    return;// trying to avoid race condition, if another thread already processed the ranking, don't store the actions again.
+  }
+  // Always fetch and merge details in the requested language
+  const mergedRanked = await fetchAndMergeRankedActions(lang, rankedActions);
+  let savedCount = 0;
+  for (const rankedAction of mergedRanked) {
+    if (!rankedAction) continue;
+    try {
+      await db.models.HighImpactActionRanked.create({
+        hiaRankingId: ranking.id,
```

Comment:
Is this the correct spelling for this key?

Reply:
yes, HIA = Hight Impact Action